### DISCUSSION TITLE
Become familiar with project stack (NestJS, NextJS, MongoDB)

### DISCUSSION THREAD

**Discussion by roberto-morado** [careerJumpstart/TestimonialFree#18] [General]
Learn the frameworks we're using and document all of your findings in here 👇 **Any and every point** you come across that strikes you as useful should be documented for future reference

**Comment by ClaraLopezUribe** [comment]
React 17 vs 18:  Which version will be used?

**Reply by roberto-morado** [reply]
React 18

**Comment by ClaraLopezUribe** [comment]
HYDRATE (in v17) is now Deprecated:

This API will be removed in a future major version of React.

In React 18, hydrate was replaced by [hydrateRoot.](https://beta.reactjs.org/reference/react-dom/client/hydrateRoot) Using hydrate in React 18 will warn that your app will behave as if it’s running React 17.

[Source/Learn more](https://beta.reactjs.org/reference/react-dom/hydrate)

**Comment by ClaraLopezUribe** [comment]
There are two types of “model” data in React: props and state. The two are very different:

[Props are like arguments you pass](https://beta.reactjs.org/learn/passing-props-to-a-component) to a function. They let a parent component pass data to a child component and customize its appearance. For example, a Form can pass a color prop to a Button.
[State is like a component’s memory.](https://beta.reactjs.org/learn/state-a-components-memory) It lets a component keep track of some information and change it in response to interactions. For example, a Button might keep track of isHovered state.

**Comment by ClaraLopezUribe** [comment]
Add state to the component with the [useState() Hook.](https://beta.reactjs.org/reference/react/useState) Hooks let you “hook into” a component’s [render cycle.](https://beta.reactjs.org/learn/render-and-commit)

You want to make it so whenever the user changes the form inputs, the state updates to reflect those changes. So, you will add the `onChange` event handlers and set the parent state from them.

**Comment by roberto-morado** [comment]
As of right now; the backend **requires** that you install an instance of MongoDB on your local machine in order to test the API locally; you can follow the [official documentation](https://www.mongodb.com/docs/manual/installation/) to install the Community Edition.

In the future, we could set up a cloud version of MongoDB and in this manner, we can avoid having to install it locally, or on the server when we launch production.

**Comment by ClaraLopezUribe** [comment]
These articles may be useful to a beginner as a road-map of key concepts to go from JavaScript -->React -->Next JS

https://www.robinwieruch.de/javascript-fundamentals-react-requirements/
https://nextjs.org/learn/foundations/about-nextjs
